build/ci_utils/src/actions/artifacts.rs

```diff
@@ -86,6 +86,7 @@ pub fn upload_single_file(
 ) -> impl Future<Output = Result> {
     let file = file.into();
     let files = single_file_provider(file);
+    #[allow(clippy::redundant_closure_call)]
```

Comment:
does it work without the closure? Why are we doing it in such a strange way? A docs explaining that should be added here IMO

Reply:
That's a really weird pattern that causes clippy lint to trigger. I'd rather refactor that than add docs, especially when it's only two lines in the whole codebase. As far as I understand, the closure comes because we want to use `?` to handle error in `files` variable, and removing closure is not possible because this function doesn't return `Result`. @mwu-tow your inside can be helpful here.